src/component.rs

```diff
@@ -144,6 +144,38 @@ pub(crate) fn get_component_update(
     }
 }
 
+#[context("query adopt impl")]
```

Comment:
The intention for `context` is to provide something friendly for an external user. Maybe `Querying adoptable state`?